### ISSUE TITLE
[amp-selector] keyboard focuses hidden options

### ISSUE THREAD

**Issue by sebastianbenz** [ampproject/amphtml#19914]
Steps to reproduce:

1. goto:https://ampbyexample.com/advanced/combobox/preview/
2. enter `g` in the search field
3. press 2x tab to focus the suggest list
4. use key down to navigate through the suggestions

Actual behaviour: the keyboard will focus hidden elements while cycling through the items
Expected behaviour: hidden elements will be ignored 

//cc @aghassemi @cvializ

**Comment by cvializ** [comment]
Oof, I can reproduce. We'll fix this one soon

**Comment by cvializ** [comment]
@sebastianbenz I propose a modification of the example: there are many many ways to hide an element, but the `hidden` attribute is a standard ways that AMP and other HTML frameworks can hide elements. Can the example use the `hidden` attribute instead of a `.hidden` class? Then the `amp-selector` extension can skip options with the `hidden` attribute when interacting with the keyboard. That strategy will allow us to avoid parsing through the many different ways that elements can become hidden, like `.hidden, .other-random-class { display: none }`

What do you think?

**Comment by sebastianbenz** [comment]
That makes sense. I'm rewriting the sample right now anyway.

>

**Comment by aghassemi** [comment]
@cvializ Runtime could skip if `height == 0` and `width == 0`. Won't cover `visibility:hidden` or `opacity:0` but a good start.

**Comment by wassgha** [comment]
@sebastianbenz could you please fix the demo for this one?

**Comment by sebastianbenz** [comment]
Hm - the demo fails because amp-autocomplete is not wrapped in an amp-form: 

```
error.js:195 amp-autocomplete should be inside a <form> tag
```

This should be a validation error. Is it possible that this got introduced later causing the demo to break?

**Comment by cvializ** [comment]
Reopening since the bug is not fixed
